### PR TITLE
Update to the helm templates to allow nodeport numbers to be defined in values.

### PR DESCRIPTION
**Pull request by tmojlupo**
Adding logic to 8-gateway-proxy-service.yaml to use nodeport numbers if predefined in the values.yml.  This would satisfy issue #2899.
https://github.com/solo-io/gloo/issues/2899

It can be used as follows in a 
gloo:
  gatewayProxies:
    gatewayProxy:
      service:
        httpPort: 80
        httpNodePort: 30001
        httpsPort: 443
        httpsNodePort: 30002
        type: NodePort
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2899

### CODE REVIEW

**Comment by tmojlupo** [comment]
> You'll need to add a new changelog file

Added with commit e143ce0.

**Comment by tmojlupo** [comment]
I updated the location of the changelog file to v1.4.0-beta9, updated the values.go file for the documentation and ran the "make update-deps generated-code -B" command to generate the updated values.txt document.

**Comment by kdorosh** [comment]
/kick regression test flake https://console.cloud.google.com/cloud-build/builds/7f6939f4-c0c1-41ac-b39f-b05d40c3c175?project=solo-public

File: install/helm/gloo/templates/8-gateway-proxy-service.yaml [hunk 1]
```diff
@@ -27,19 +27,31 @@ spec:
     targetPort: {{ $spec.podTemplate.httpsPort }}
     protocol: TCP
     name: https
+    {{- if and (eq $spec.service.type "NodePort") $spec.service.httpsNodePort }}
```
Comment (yuval-k) [inline install/helm/gloo/templates/8-gateway-proxy-service.yaml:30]: load balancer services also have a nodeport

**Pull request merged**